cdap-ui/app/cdap/components/DataPrep/DataPrepServiceControl/index.js

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import React, { Component } from 'react';
+import MyDataPrepApi from 'api/dataprep';
+import {MyArtifactApi} from 'api/artifact';
+import find from 'lodash/find';
+import NamespaceStore from 'services/NamespaceStore';
+
+export default class DataPrepServiceControl extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      loading: false,
+      error: null,
+      extendedMessage: null
+    };
+
+    this.enableService = this.enableService.bind(this);
+  }
+
+  componentWillUnmount() {
+    if (this.servicePoll && this.servicePoll.dispose) {
+      this.servicePoll.dispose();
+    }
+  }
+
+  enableService() {
+    this.setState({loading: true});
+
+    /**
+     *  1. Get Wrangler Service App
+     *  2. If not found, create app
+     *  3. Start Wrangler Service
+     *  4. Poll until service starts, then reload page
+     **/
+
+    let namespace = NamespaceStore.getState().selectedNamespace;
+
+    MyDataPrepApi.getApp({ namespace })
+      .subscribe(() => {
+        // Wrangler app already exist
+        // Just start service
+        this.startService();
+      }, () => {
+        // App does not exist
+        // Go to create app
+        this.createApp();
+      });
+  }
+
+  createApp() {
+    let namespace = NamespaceStore.getState().selectedNamespace;
+
+    MyArtifactApi.list({ namespace })
+      .subscribe((res) => {
+        let artifact = find(res, { 'name': 'wrangler-service' });
+
+        MyDataPrepApi.createApp({ namespace }, { artifact })
+          .subscribe(() => {
+            this.startService();
+          }, (err) => {
+            this.setState({
+              error: 'Failed to start service',
```

Comment:
This error is surfacing to the user, they have no idea that this is creating an app behind the scenes. They are just enabling the service.